explorer/server.go

```diff
@@ -0,0 +1,218 @@
+package explorer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg/rmb"
+)
+
+func (a *App) runServer() {
+	log.Info().Str("Server started ... listening on", string(a.explorer)).Msg("")
+
+}
+
+func (a *App) listFarms(w http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("500 - Something bad happened!"))
+		return
+	}
+	log.Debug().Str("request_body", string(body)).Msg("request from external agent")
+	queryString := `
+	{
+		farms {
+			name
+			farmId
+			twinId
+			version
+			farmId
+			pricingPolicyId
+		}
+		publicIps{
+			id
+			ip
+			farmId
+			contractId
+			gateway
+			
+		}
+	}
+	`
+	farmsData := query(queryString)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("500 - Something bad happened!"))
+	}
+	jsonBytes := []byte(farmsData)
+
+	w.Write(jsonBytes)
```

Comment:
why do u have to create an intermediate variable ?
```suggestion
	w.Write([]byte(farmsData))
```